src/mark.js

```diff
@@ -157,23 +158,30 @@ class Mark { // eslint-disable-line no-unused-vars
      */
     createSynonymsRegExp(str) {
         const syn = this.opt.synonyms,
-            sens = this.opt.caseSensitive ? "" : "i";
+            sens = this.opt.caseSensitive ? "" : "i",
+            // add replacement character placeholder before and after the
+            // synonym group
+            joinerPlaceholder = this.opt.ignoreJoiners ||
+                this.opt.ignorePunctuation.length ? "\u0000" : "";
```

Comment:
Yes, that is the fix for the problem I mentioned in [this comment](https://github.com/julmot/mark.js/pull/135#issuecomment-300046566)